src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/ProjectView.xaml.cs

```diff
@@ -248,10 +259,11 @@ private void SetComboboxCurrentVersion(string comboboxText, IEnumerable<NuGetVer
             for (int i = 0; i < _versions.Items.Count; i++)
             {
                 DisplayVersion currentItem = _versions.Items[i] as DisplayVersion;
-                if (currentItem != null && (comboboxText == _versions.Items[i].ToString() || _versions.Items[i].ToString() == matchVersion?.ToString()))
+                if (currentItem != null && // null, this represent a bar in UI in the Versions combobox 
+                    (comboboxText.Trim() == currentItem.Version.ToNormalizedString() || matchVersion?.ToString() == currentItem.Version.ToNormalizedString())) // trim extra spaces and compare versions without labels
                 {
                     _versions.SelectedIndex = i; // This is the "select" effect in the dropdown
-                    PackageDetailControlModel.SelectedVersion = new DisplayVersion(userRange, matchVersion, additionalInfo: null, isVulnerable: currentItem.IsVulnerable);
+                    PackageDetailControlModel.SelectedVersion = new DisplayVersion(userRange, matchVersion, additionalInfo: null, isDeprecated: currentItem.IsDeprecated, isVulnerable: currentItem.IsVulnerable);
```

Comment:
Can't we bail out of the loop once a version is selected? Does this select all matching versions? I believe what I see here is that it will select every version that matches, then the final state is the last match will be selected. If that's the desired behavior, there should be a more explicit way to do that.

Reply:
We can return once a version is selected